fuzzbench/test_e2e_run.py

```diff
@@ -26,9 +27,21 @@
 class TestEndToEndRunResults:
     """Checks the result of a test experiment run."""
 
-    def test_all_jobs_finished_sucessfully(self):
-        """Fake test to be implemented later."""
-        assert True
+    def test_jobs_dependency(self, redis_connection):
+        """Tests that jobs dependency preserves during working."""
+        jobs = {
+            name: Job.fetch(name, connection=redis_connection)
+            for name in ['base-image', 'base-builder', 'base-runner']
+        }
+        assert jobs['base-image'].ended_at <= jobs['base-builder'].started_at
+        assert jobs['base-builder'].ended_at <= jobs['base-runner'].started_at
```

Comment:
>Yeah I don't think this makes sense to test for. They are built in parallel.

So, shall we change it back? Say, base-runner depends on base-image, not base-builder?

Reply:
Yes base-runner does not depend on base-builder.